onadata/libs/models/share_project.py

```diff
@@ -28,6 +29,18 @@ def save(self, **kwargs):
 
                 # apply same role to forms under the project
                 for xform in self.project.xform_set.all():
+                    # check if there is xform meta perms set
+                    meta_perms = xform.metadata_set \
+                        .filter(data_type='xform_meta_perms')
+                    if meta_perms:
+                        meta_perm = meta_perms[0].data_value.split("|")
+
+                        if len(meta_perm) > 1:
+                            if role == EditorRole:
+                                role = ROLES.get(meta_perm[0])
+
+                            elif role == DataEntryRole:
+                                role = ROLES.get(meta_perm[1])
```

Comment:
Would it be correct to say the original role gets overridden here? Let's say the role that had been applied was not a meta-permission, this seems to override those permissions when you have a form/project that has meta-permission. Probably a test to show that this is not the case would be helpful.